### ISSUE TITLE
[Enhancement]: make the remove coupon button display optional based on a flag value in the cart state

### ISSUE THREAD

**Issue by junixdev** [woocommerce/woocommerce#44873]
### Describe the solution you'd like

Similar to how the "remove item" in the cart/checkout blocks can be modified via the `registerCheckoutFilters` hook, I believe this needs to also be possible for coupons.

Currently this is not possible based as there is no flag available for this. Each cart item has a `showRemoveItemLink` flag in the cart store, so maybe add a `showRemoveCouponButton` flag for each coupon as well.

### Describe alternatives you've considered

None.

### Additional context

We're making our plugin be compatible with the Cart/Checkout blocks and we have a feature that auto applies the coupon to the cart. In the classic cart/checkout, the "remove" button is not displayed as it doesn't make sense UX wise to show this as removing the coupon manually will just add it back when the cart data refreshes.

**Comment by opr** [comment]
Hi, thanks for the suggestion. There are multiple parts to your request because simply hiding the button will not prevent the coupon removal (it is possible to remove via store API). 

I'm wondering if you want to totally prevent removal or simply just want to hide the button?

If the latter, I wonder if we added an attribute `data-coupon-code` to the coupon which you could use to to hide the button with CSS would be sufficient? It would open up more coupon-specific styling use cases too.

If the former, then we would likely have to look into extending the WC Core functionality.

Thanks again for opening this issue, I'm adding this to our backlog. Any further feedback you have on the questions above would be helpful.

**Comment by junixdev** [comment]
@opr I think we will only need the functionality to hide the button, but it would also be good though to have some filters available  in the `WC_Cart::remove_coupon()` and `WC_Cart::remove_coupons()` methods, so we can control which coupons shouldn't be removable.

For now though, having the ability to hide the button will suffice for our use case so we can have a similar UX with the classic cart/checkout for our plugin's auto apply feature.

For hiding the button, I'd prefer to do it similar to how we can hide the "remove item" button for cart items via filtering the `showRemoveItemLink` flag parameter. Another way is probably to allow us to append a custom classname for each coupon card.

thanks a bunch!

**Comment by opr** [comment]
Thanks for the additional information, I'm adding this to the backlog.

I think the code changes for removing the button would be fairly simple so if this is urgent for you then you may consider opening a PR. I would be able to help with reviews etc. if you choose to contribute! 🙂

**Comment by pierorocca** [comment]
@yjailin, I think the ask here is reasonable and wanted you to sanity check this scenario where a user would not be able to remove an auto applied coupon.

**Comment by yjailin** [comment]
@pierorocca Yes, that's fine. If the shopper qualifies, removing the discount does nothing, it just comes back on the next refresh. Codes they typed stay removable. If we want to support this case, the order summary needs to support both removable discounts and auto-applied/non removable discounts.

One idea could be to drop the discount chips and add a line per discount. Here is a mockup with 1 auto applied/non removable discount and 2 applied/removable coupons.

<img width="378" height="677" alt="Image" src="https://github.com/user-attachments/assets/62961ee4-1f4c-45f2-9032-b06a0c94d22f" />
